ring_test.go

```diff
@@ -172,6 +172,34 @@ var _ = Describe("Redis Ring", func() {
 			Expect(ringShard2.Info().Val()).To(ContainSubstring("keys=100"))
 		})
 	})
+
+	Describe("shard passwords", func() {
+		It("can be initialized with a single password, used for all shards", func() {
+			ring = redis.NewRing(&redis.RingOptions{
+				Addrs: map[string]string {
+					"shard1": "host1",
+				},
+				Password: "some-password",
+			})
+
+			Expect(ring).ToNot(BeNil())
```

Comment:
ring can never be nil here. You should at least call `ring.Ping()` to test if this is working.

Reply:
:+1: